src/blockchain/shelley/utils.js

```diff
@@ -130,16 +140,19 @@ const fragmentToObj = (fragment: any, extraData: {}): TxType => {
         }
         const deleg = cert.get_owner_stake_delegation()
         const poolId = deleg.delegation_type().get_full()
-        common.certificate = {
-          type: 'OwnerStakeDelegation',
+        const parsedCert: StakeDelegationType = {
+          type: CERT_TYPE.StakeDelegation,
           // TODO: possibly handle Ratio types
           pool_id: poolId != null ? poolId.to_string() : null,
           account: inputs_parsed[0].account_id,
+          isOwnerStake: true,
         }
+        common.certificate = parsedCert
         break
       }
       default:
-        throw new Error(`parsing certificate type not implemented: ${cert.get_type()}`)
+        break
```

Comment:
Which certificate types do we not support? We should at least log this as a warn level or something so we know.